nixos/modules/system/boot/unl0kr.nix

```diff
@@ -80,68 +90,90 @@ in
         "wacom"
       ];
 
-    boot.initrd.systemd = {
-      contents."/etc/unl0kr.conf".source = settingsFormat.generate "unl0kr.conf" cfg.settings;
-      storePaths = with pkgs; [
-        "${pkgs.gnugrep}/bin/grep"
-        libinput
-        xkeyboard_config
-        "${config.boot.initrd.systemd.package}/lib/systemd/systemd-reply-password"
-        (lib.getExe' cfg.package "unl0kr")
-      ];
-      services = {
-        unl0kr-ask-password = {
-          description = "Forward Password Requests to unl0kr";
-          conflicts = [
-            "emergency.service"
-            "initrd-switch-root.target"
-            "shutdown.target"
-          ];
-          unitConfig.DefaultDependencies = false;
-          after = [
-            "systemd-vconsole-setup.service"
-            "udev.service"
-          ];
-          before = [ "shutdown.target" ];
-          script = ''
-            # This script acts as a Password Agent: https://systemd.io/PASSWORD_AGENTS/
-
-            DIR=/run/systemd/ask-password/
-            # If a user has multiple encrypted disks, the requests might come in different times,
-            # so make sure to answer as many requests as we can. Once boot succeeds, other
-            # password agents will be responsible for watching for requests.
-            while [ -d $DIR ] && [ "$(ls -A $DIR/ask.*)" ];
-            do
-              for file in `ls $DIR/ask.*`; do
-                socket="$(cat "$file" | ${pkgs.gnugrep}/bin/grep "Socket=" | cut -d= -f2)"
-                ${lib.getExe' cfg.package "unl0kr"} -v -C "/etc/unl0kr.conf" | ${config.boot.initrd.systemd.package}/lib/systemd/systemd-reply-password 1 "$socket"
+    boot.initrd.systemd = lib.mkMerge [
+      {
+        contents."/etc/unl0kr.conf".source = settingsFormat.generate "unl0kr.conf" cfg.settings;
+        storePaths = lib.mkMerge ([
+          (with pkgs; [
+            libinput
+            libwacom
+            xkeyboard_config
+          ])
+
+          (lib.mkIf (!cfg.useUpstreamPasswordAgent) [
+            "${pkgs.gnugrep}/bin/grep"
+            "${config.boot.initrd.systemd.package}/lib/systemd/systemd-reply-password"
+            (lib.getExe' cfg.package "unl0kr")
+          ])
+
+          (lib.mkIf (cfg.useUpstreamPasswordAgent) [ cfg.package ])
```

Comment:
It should just be the paths actually needed, i.e. `unl0kr` and `unl0kr-agent`.

Reply:
I do not understand

Reply:
Do you mean just putting unl0kr? Could check it out right now